src/coreclr/classlibnative/bcltype/system.cpp

```diff
@@ -143,7 +145,10 @@ extern "C" void QCALLTYPE Environment_FailFast(QCall::StackCrawlMarkHandle mark,
 
     LPCWSTR argExceptionString = NULL;
     StackSString msg;
-    if (exception.Get() != NULL)
+    // because Environment_FailFast should kill the process, any subsequent calls are likely nested call from managed while unwinding the stack.
```

Comment:
This is going to skip formatting the error message if there are multiple threads crashing at the same time and racing to which one of them is going to crash first.

I think the counter should be thread_local (and possibly just `int8_t`).